### PR TITLE
Stasis Beds

### PR DESCRIPTION
**Pull request by Geevies**
* Ported stasis beds. RnD can print their circuitboards once they've reached level four research in magnets, engineering and bio.

Ported from: /tg/station

image

### CODE REVIEW

**Comment by Geevies** [comment]
!review

File: code/game/machinery/stasis_bed.dm [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+		else
+			playsound(src, 'sound/machines/synth_no.ogg', 50, TRUE, frequency = sound_freq)
+		last_stasis_sound = _running
+
+/obj/machinery/stasis_bed/AltClick(mob/user)
+	if((world.time >= stasis_can_toggle) && !use_check_and_message(user) && !(stat & (NOPOWER|BROKEN)))
+		stasis_enabled = !stasis_enabled
+		stasis_can_toggle = world.time + 5 SECONDS
+		playsound(src, 'sound/machines/click.ogg', 60, TRUE)
+		user.visible_message("<span class='notice'>\The [src] [stasis_enabled ? "powers on" : "shuts down"].</span>", \
```
Comment (JohnWildkins) [inline code/game/machinery/stasis_bed.dm:57]: `SPAN_NOTICE`

File: code/game/machinery/stasis_bed.dm [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+		/obj/item/circuitboard/stasis_bed,
+		/obj/item/stock_parts/micro_laser = 1,
+		/obj/item/stock_parts/capacitor = 2,
+		/obj/item/stock_parts/scanning_module = 2,
+		/obj/item/stock_parts/console_screen
+	)
+
+/obj/machinery/stasis_bed/Initialize(mapload, d, populate_components)
+	. = ..()
+	LAZYADD(can_buckle, /mob/living/carbon/human)
```
Comment (JohnWildkins) [inline code/game/machinery/stasis_bed.dm:30]: `can_buckle` is not a lazy list

**Review by JohnWildkins — APPROVED** [review]

File: code/_helpers/unsorted.dm [hunk 3]
```diff
@@ -828,7 +828,7 @@ var/global/list/common_tools = list(
 /proc/can_operate(mob/living/carbon/M) //If it's 2, commence surgery, if it's 1, fail surgery, if it's 0, attack
 	var/surgery_attempt = SURGERY_IGNORE
 	var/located = FALSE
-	if(locate(/obj/machinery/optable, M.loc))
+	if(istype(M.buckled_to, /obj/machinery/stasis_bed) || locate(/obj/machinery/optable, M.loc))
```
Comment (NonQueueingMatt) [inline code/_helpers/unsorted.dm:831]: I hate this. Like, a real lot.

**Review by NonQueueingMatt — APPROVED** [review]

File: code/_helpers/unsorted.dm (hunk 3 above)
Comment (Geevies) [reply code/_helpers/unsorted.dm:831]: honey it's 3 am time for your surgery code refactor

**Pull request merged**